src/dssgsolve/settings.py

```diff
@@ -20,13 +21,31 @@
 BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
 
 SITE_NAME=config('SITE_NAME', default='DSSG Solve')
+
 # Quick-start development settings - unsuitable for production
 # See https://docs.djangoproject.com/en/2.0/howto/deployment/checklist/
 
 SECRET_KEY = config('SECRET_KEY')
-DEBUG = config('DEBUG', default = False, cast = bool)
 
-ALLOWED_HOSTS = config('ALLOWED_HOSTS', cast = Csv())
+DEBUG = config('DEBUG', default=False, cast=bool)
+
+ALLOWED_HOSTS = config('ALLOWED_HOSTS', cast=Csv())
+
+if DEBUG:
+    EC2_PRIVATE_IP = None
+else:
+    try:
+        response = requests.get(
+            'http://169.254.169.254/latest/meta-data/local-ipv4',
+            timeout=0.01
+        )
+    except requests.exceptions.RequestException:
+        EC2_PRIVATE_IP = None
+    else:
+        EC2_PRIVATE_IP = response.text
+
+    if EC2_PRIVATE_IP:
+        ALLOWED_HOSTS.append(EC2_PRIVATE_IP)
```

Comment:
I like this the least by far, but it seemed like the most expedient solution to the problem of the load balancer running HTTP request health checks against the webapp, (of course without the header `Host: solve.dssg.io`, but rather just at the web server's IP address).

If the webapp isn't configured to allow the load balancer's requests, it returns a non-200 response, which the load balancer perceives as an indication that the container is "unhealthy" and should be replaced. Until I put this in, the container was being constantly created, started, tested, stopped, destroyed and recreated….

So, upon process (and app) initialization, the app makes a call to an AWS metadata server, to learn its private IP. Then, it can allow requests to this IP.

We could perhaps make this less dynamic, but considering the ephemeral nature of EC2 instances, we couldn't make it all that static/global. For example, we could specify a script to run at EC2 instance initialization, which would cache the private IP. (Or, it *might* be safe to infer the private IP from the Docker host's hostname.) But, I'm not itching to get into that.